source/h2_stream.c

```diff
@@ -242,8 +249,6 @@ struct aws_h2_stream *aws_h2_stream_new_request(
     /* Stream refcount starts at 1, and gets incremented again for the connection upon a call to activate() */
     aws_atomic_init_int(&stream->base.refcount, 1);
 
-    /* Init H2 specific stuff */
-    stream->thread_data.state = AWS_H2_STREAM_STATE_IDLE;
     enum aws_http_version message_version = aws_http_message_get_protocol_version(options->request);
     switch (message_version) {
         case AWS_HTTP_VERSION_1_1:
```

Comment:
unrelated to this PR:
Can we add a TODO to get rid of automatic transformation of HTTP/1 messages? It seems much better to have this be explicit. We discussed earlier today how the transformation may need to know more information that we don't have available right now (ex: schema)